src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -0,0 +1,75 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { INTERNAL_USE_ONLY } from '../../../common/http-request';
+import { ApproveDataOptions, PaypalCommerceScriptLoader } from '../../../payment/strategies/paypal-commerce';
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _formPoster: FormPoster,
+        private _requestSender: RequestSender
+    ) {}
+
+    initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+        const paypalOptions = options.paypalCommerce;
+
+        if (!paypalOptions || !paypalOptions.clientId) {
+            throw new InvalidArgumentError();
+        }
+
+        return this._paypalScriptLoader.loadPaypalCommerce(paypalOptions.clientId)
+            .then(paypal => {
+                return paypal.Buttons({
+                    createOrder: () => this._setupPayment(),
+                    onApprove: data => this._tokenizePayment(paymentMethod.id, data),
+                }).render(`#${options.containerId}`);
+            });
+    }
+
+    deinitialize(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    private _setupPayment(): Promise<string> {
+
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
+            .then(state => {
+                const url = '/api/storefront/payment/paypalcommerce';
+                const headers = {
+                    'X-API-INTERNAL': INTERNAL_USE_ONLY,
+                    'Content-Type': 'application/x-www-form-urlencoded',
```

Comment:
I'm still not sure why we need to post the content type as `application/x-www-form-urlencoded`. The body should be passed in as a `JsonBody` as an argument for the controller action. @BC-ACherednichenko can you take a look at how other post/put actions are written?

Reply:
Thank you!
I removed the doc file and canceled the changes in package-lock.json

Reply:
@davidchin, this part of the functionality is consistent with PayPal express `api/storefront/payment/paypalexpress`, we decided to use this route `api/storefront/payment/{storefront}` instead of creating a new one, because and it does the same thing as PayPalCommerce does, once PayPal express eliminated we can rebuild this controller to accept JSON. 
 This route, in general, accepts data in form-urlencoded format, but in comparison with PayPal Commerce implementation, it sends a request differently 

 https://github.com/bigcommerce/checkout-sdk-js/blob/8044df22b714020444a523e96ee96b7c89b010b8/src/checkout-buttons/strategies/paypal/paypal-button-strategy.ts#L89

```js
        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
            .then(state => {
                const cart = state.cart.getCart();
                const cartId = cart ? cart.id : '';

                return actions.request.post(`${this._host}/api/storefront/payment/paypalexpress`, { merchantId, cartId }, {
                    headers: {
                        'X-API-INTERNAL': INTERNAL_USE_ONLY,
                    },
                });
            })
```

Reply:
I'm a little bit confused. 🤔If PayPal express doesn't require a different content-type and it uses the same route / controller, why does PayPal Commerce require something different?

Reply:
per our conversation
in PayPal commerce we send a request with `RequestSender` - our own library,   in PayPal express it being sent thru actions object that passed from PayPal side on SDK side, it has different params
https://github.com/bigcommerce/checkout-sdk-js/blob/71980ef5ec64693851a1b8856ca559df091c12ea/src/payment/strategies/paypal/paypal-sdk.ts#L56